src/plugins/data/public/search/legacy/default_search_strategy.ts

```diff
@@ -30,34 +31,47 @@ export const defaultSearchStrategy: SearchStrategyProvider = {
   },
 };
 
-function msearch({
-  searchRequests,
-  legacySearchService,
-  config,
-  esShardTimeout,
-}: SearchStrategySearchParams) {
-  const es = legacySearchService.esClient;
-  const inlineRequests = searchRequests.map(({ index, body, search_type: searchType }) => {
-    const inlineHeader = {
-      index: index.title || index,
-      search_type: searchType,
-      ignore_unavailable: true,
-      preference: getPreference(config.get),
-    };
-    const inlineBody = {
-      ...body,
-      timeout: getTimeout(esShardTimeout),
+function msearch({ searchRequests, config, http }: SearchStrategySearchParams) {
+  const requests = searchRequests.map(({ index, body }) => {
+    return {
+      header: {
+        index: index.title || index,
+        preference: getPreference(config.get),
+      },
+      body,
     };
-    return `${JSON.stringify(inlineHeader)}\n${JSON.stringify(inlineBody)}`;
   });
 
-  const searching = es.msearch({
-    ...getMSearchParams(config.get),
-    body: `${inlineRequests.join('\n')}\n`,
+  const abortController = new AbortController();
+  let resolved = false;
+
+  const loadingCount$ = new BehaviorSubject(0);
+  http.addLoadingCountSource(loadingCount$);
```

Comment:
I don't think you want to add a new `BehaviorSubject` for every msearch. Especially as there's no way to un-register them.

This should be managed at a higher level, passing in the `loadingCount$` to be used.

Reply:
Moved `loadingCount$` to search service